app/contracts/project_custom_field_project_mappings/base_contract.rb

```diff
@@ -46,7 +46,7 @@ def not_required
       # enabling a custom field which is required happens in an after_save hook within the custom field model itself
       return if model.project_custom_field.nil? || !model.project_custom_field.required?
 
-      errors.add :custom_field_id, :invalid
+      errors.add :custom_field_id, :cannot_delete_mapping
```

Comment:
The error messages were rather generic and not directly helpful to the user: "Custom field Id is invalid" was what was present before